File: src/main/java/de/SparkArmy/utils/LogChannelType.java
package de.SparkArmy.utils;

@SuppressWarnings("unused")
public enum LogChannelType {
    MESSAGE(1,"message-log"),
    MEMBER (2,"member-log"),
    COMMAND (3,"command-log"),
    LEAVE (4,"leave-log"),
    MOD (5,"mod-log"),
    SERVER (6,"server-log"),
    VOICE (7,"voice-log");

    private final Integer id;
    private final String name;


    LogChannelType(Integer logId, String name) {
        id = logId;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: src/main/java/de/sparkarmy/model/LogChannelType.kt
package de.sparkarmy.model

import de.sparkarmy.data.BitField

enum class LogChannelType(override val offset: Int, val identifier: String) : BitField {
    MESSAGE_LOG(0, "Message-Log"),
    MEMBER_LOG(1, "Member-Log"),
    COMMAND_LOG(2, "Command-Log"),
    SERVER_LOG(3, "Server-Log"),
    VOICE_LOG(4, "Voice-Log"),
    MOD_LOG(5, "Mod-Log"),
    LEAVE_LOG(6, "Leave-Log"),
}